File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/Java/C/s380356166_comments_removed.c
#include <stdio.h>
#include <stdlib.h>

typedef struct {
    int key;
    int value;
} Pair;

int nextInt() {
    int c, n = 0;
    while ((c = getchar()) != EOF && c != '\n' && c != ' ');
    while ((c = getchar()) != EOF && c != '\n' && c != ' ') {
        n = n * 10 + (c - '0');
    }
    return n;
}

void solve(int K, int S) {
    int ans = 0;
    int wk = 0;
    for (int x = 0; x <= K; x++) {
        for (int y = 0; y <= K; y++) {
            wk = S - x - y;
            if (wk >= 0 && wk <= K) {
                ans++;
            }
        }
    }
    printf("%d\n", ans);
}

int main() {
    int K = nextInt();
    int S = nextInt();
    solve(K, S);
    return 0;
}

// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/Java/Go/s380356166_comments_removed.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

type Solver struct{}

func (s *Solver) solve(testNumber int, in *MyInput, out *bufio.Writer) {
	ans := 0
	K := in.nextInt()
	S := in.nextInt()
	wk := 0
	for x := 0; x <= K; x++ {
		for y := 0; y <= K; y++ {
			wk = S - x - y
			if wk >= 0 && wk <= K {
				ans++
			}
		}
	}
	fmt.Fprintln(out, ans)
	out.Flush()
}

type MyInput struct {
	scanner *bufio.Scanner
}

func NewMyInput(reader *bufio.Reader) *MyInput {
	return &MyInput{scanner: bufio.NewScanner(reader)}
}

func (in *MyInput) nextInt() int {
	in.scanner.Scan()
	num, _ := strconv.Atoi(strings.TrimSpace(in.scanner.Text()))
	return num
}

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	myInput := NewMyInput(reader)
	solver := &Solver{}
	solver.solve(1, myInput, writer)
}

// End of Code
